### PR TITLE
Passing events to sub-widgets in List Editor and refactoring of the slider component.

### LINKED ISSUE #6428: `Vector_Editor` component allowing passing events to sub-widgets.
This issue is about extending Vector Editor with the ability to pass events to its sub-widgets. For example, sliders in Vector Editor should get proper mouse events when Vector Editor discovers they are not being dragged.

### PR DESCRIPTION
**Pull request by wdanilo**
### Pull Request Description

Fixes https://github.com/enso-org/enso/issues/6428 .
This PR improves List Editor component and refactors the Slider component.

### List Editor Improvements
It is possible now to use complex widgets as List Editor items (previously dragging them was not possible). Also, now the mouse events are passed to the items if the item is not being dragged:

https://user-images.githubusercontent.com/1623053/234589248-a5256b20-59ef-4eb6-9b92-fa6b28c9340a.mov

### Slider Improvements
The slider code was heavily refactored to make it use the most modern EnsoGL API. In particular:
- The track / thumb shape was unified, so less shapes are used.
- The slider size is now controlled by display objects, rather than manually controlled by FRP endpoints.
- The slider start, middle, and end part of the track can be dragged now.
- The slider origin is now in it's left-bottom corner.

https://user-images.githubusercontent.com/1623053/234591547-6b2a056b-526b-4173-bb65-a9e56c3f3ff6.mov

However, the slider code after the update do not contain all the functionality of the old implementation. In particular, the following things do not work yet:
- Vertical sliders.
- Overflow values.

As the slider is not used in real places, we are keeping it as is for now, while the created changes were needed to test it's behavior with List Editor component.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: lib/rust/ensogl/component/list-editor/src/lib.rs [hunk 1]
```diff
@@ -262,12 +258,12 @@ impl<T> From<StrongPlaceholder> for ItemOrPlaceholder<T> {
...
 
-ensogl_core::define_endpoints_2! { <T: ('static)>
+ensogl_core::define_endpoints_2! { <T: ('static + Debug)>
     Input {
         /// Push a new element to the end of the list.
-        push(Weak<T>),
+        push(Rc<RefCell<Option<T>>>),
 
-        insert((Index, Weak<T>)),
+        insert((Index, Rc<RefCell<Option<T>>>)),
```
Comment (mwu-tow) [inline lib/rust/ensogl/component/list-editor/src/lib.rs:268]: Why this has no docstring while other two inputs do?

File: lib/rust/ensogl/component/list-editor/src/lib.rs [hunk 2]
```diff
@@ -362,7 +358,7 @@ impl<T: display::Object + CloneRef> ListEditor<T> {
 
             // Do not pass events to children, as we don't know whether we are about to drag
             // them yet.
-            eval on_down ([] (event) event.stop_propagation());
+
```
Comment (mwu-tow) [inline lib/rust/ensogl/component/list-editor/src/lib.rs]: Does the comment above remains up-to-date after the change?

File: lib/rust/ensogl/component/list-editor/src/lib.rs [hunk 3]
```diff
@@ -283,9 +279,9 @@ ensogl_core::define_endpoints_2! { <T: ('static)>
     }
     Output {
         /// Fires whenever a new element was added to the list.
-        on_item_added(Response<(Index, Weak<T>)>),
+        on_item_added(Response<Index>),
 
-        on_item_removed(Response<(Index, Weak<T>)>),
+        on_item_removed(Response<(Index, Rc<RefCell<Option<T>>>)>),
```
Comment (mwu-tow) [inline lib/rust/ensogl/component/list-editor/src/lib.rs:304]: The same question as above.

File: lib/rust/ensogl/examples/slider/src/lib.rs [hunk 4]
```diff
@@ -91,107 +75,107 @@ impl Model {
...
-        slider10.set_y(-35.0);
-        slider10.set_x(430.0);
-        slider10.frp.set_value_indicator_color(color::Lcha(0.4, 0.7, 0.7, 1.0));
-        slider10.frp.show_value(false);
-        slider10.frp.set_precision_adjustment_disabled(true);
-        slider10.frp.kind(slider::Kind::Scrollbar(0.1));
-        slider10.frp.orientation(Axis2::Y);
-        self.root.add_child(&slider10);
-        self.sliders.borrow_mut().push(slider10);
+        //
```
Comment (mwu-tow) [inline lib/rust/ensogl/examples/slider/src/lib.rs:83]: Tons of commented-out code. Is it needed? If so, why is it commented?

File: lib/rust/ensogl/examples/list-editor/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    let root = display::object::Instance::new();
+    root.set_size(Vector2(300.0, 100.0));
+    root.add_child(&vector_editor);
+    world.add_child(&root);
+
+    world.keep_alive_forever();
+    mem::forget(frp);
+    mem::forget(navigator);
+    mem::forget(root);
+    mem::forget(vector_editor);
```
Comment (mwu-tow) [inline lib/rust/ensogl/examples/list-editor/src/lib.rs:92]: If lifetimes of these objects are tied, shouldn't they be grouped together in a single struct?

**Review by mwu-tow — APPROVED** [review]

File: lib/rust/ensogl/examples/slider/src/lib.rs (hunk 4 above)
Comment (wdanilo) [reply lib/rust/ensogl/examples/slider/src/lib.rs:83]: This is explained in the PR description and in the file - the commented code needs to be uncommented and updated one day. For now, the slider implementation is improved in some places, but some places from the old implementation are not working yet.

File: lib/rust/ensogl/examples/list-editor/src/lib.rs (hunk 5 above)
Comment (wdanilo) [reply lib/rust/ensogl/examples/list-editor/src/lib.rs:92]: As agreed via Discord, let's leave it as it. It is a debug scene and we are doing so there as this is less boilerplate than creating structures. But generally, yes, we should think about managing that nicer in debug scenes in general.

**Pull request merged**